project-generator/src/main/java/io/syndesis/project/converter/DefaultProjectGenerator.java

```diff
@@ -147,41 +155,82 @@ private Mustache compile(ProjectGeneratorProperties generatorProperties, String
             }
 
             try {
-                contents.put(additionalResource.getDestination(), Files.readAllBytes(Paths.get(resource.toURI())));
+                addTarEntry(tos, additionalResource.getDestination(), Files.readAllBytes(Paths.get(resource.toURI())));
             } catch (URISyntaxException e) {
                 throw new IOException(e);
             }
         }
+    }
+
+
+    private InputStream createTarInputStream(GenerateProjectRequest request) throws IOException {
+        PipedInputStream is = new PipedInputStream();
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        PipedOutputStream os = new PipedOutputStream(is);
+        executor.submit(generateAddProjectTarEntries(request, os));
 
-        contents.put("README.md", generateFromRequest(request, readmeMustache));
-        contents.put("src/main/java/io/syndesis/example/Application.java", generateFromRequest(request, applicationJavaMustache));
-        contents.put("src/main/resources/application.properties", generateFromRequest(request, applicationPropertiesMustache));
-        contents.put("src/main/resources/syndesis.yml", generateFlowYaml(contents, request));
-        contents.put("pom.xml", generatePom(request.getIntegration()));
+        return is;
+    }
+
+    private Runnable generateAddProjectTarEntries(GenerateProjectRequest request, PipedOutputStream os) {
```

Comment:
Could be inner class that implements `Callable`, so that `IOException` can be thrown, also https://stackoverflow.com/questions/33686430/exception-propagation-within-pipedinputstream-and-pipedoutputstream